scripts/degenApe.js

```diff
@@ -115,301 +74,123 @@ const executeTx = async (calls, tx, fn, ...args) => {
-      console.log(`✔️ Strategy deployed at: ${txRefs['strategy'].address} `);
+      console.log(`Deploying ${txRefs.'name'}...`);
+      txRefs.'strategy' = await executeTx(callAttempts, 'strategy', StrategyFactory.deploy.bind(StrategyFactory), governance, strategist, controller, timelock);
+      console.log(`✔️ Strategy deployed at: ${ txRefs.'strategy'.address } `);
 
       // Get Want
-      await sleep(sleepTime, sleepToggle);
-      txRefs['want'] = await txRefs['strategy'].want();
-
-      // Deploy PickleJar contract
-      await executeTx(callAttempts, 'jar', PickleJarFactory.deploy.bind(PickleJarFactory), txRefs['want'], governance, timelock, controller);
-      console.log(`✔️ PickleJar deployed at: ${txRefs['jar'].address} `);
+      await sleep(sleepConfigs);
+      txRefs.'want' = await txRefs.'strategy'.want();
 
       // Log Want
-      console.log(`Want address is: ${txRefs['want']} `);
-      console.log(`Approving want token for deposit...`);
-      await sleep(sleepTime, sleepToggle);
-      txRefs['wantContract'] = await ethers.getContractAt("ERC20", txRefs['want']);
-
-      // Approve Want
-      await executeTx(callAttempts, 'approveTx', txRefs['wantContract'].approve, txRefs['jar'].address, ethers.constants.MaxUint256);
-      console.log(`✔️ Successfully approved Jar to spend want`);
-      console.log(`Setting all the necessary stuff in controller...`);
+      console.log(`Want address is: ${ txRefs.'want' } `);
+      await sleep(sleepConfigs);
+      txRefs.'wantContract' = await ethers.getContractAt("ERC20", txRefs.'want');
+
+      // Check if Want already has a Jar on Controller
+      await sleep(sleepConfigs);
+      txRefs.'jar' = await Controller.jars(txRefs.'want');
+
+      if (!txRefs.'jar') {
+        // Deploy PickleJar contract
+        txRefs.'jar' = await executeTx(callAttempts, 'jar', PickleJarFactory.deploy.bind(PickleJarFactory), txRefs.'want', governance, timelock, controller);
+        console.log(`✔️ PickleJar deployed at: ${ txRefs.'jar'.address } `);
+
+        // Set Jar
+        txRefs.'setJarTx' = await executeTx(callAttempts, 'setJarTx', Controller.setJar, txRefs.'want', txRefs.'jar'.address);
+        console.log(`Jar Set!`);
+
+        // Approve Want
+        console.log(`Approving want token for deposit...`);
+        txRefs.'approveTx' = await executeTx(callAttempts, 'approveTx', txRefs.'wantContract'.approve, txRefs.'jar'.address, ethers.constants.MaxUint256);
+        console.log(`✔️ Successfully approved Jar to spend want`);
+      } else {
+        console.log(`Jar for this want already exists`);
+      }
 
       // Approve Strategy
-      await executeTx(callAttempts, 'approveStratTx', Controller.approveStrategy, txRefs['want'], txRefs['strategy'].address);
+      txRefs.'approveStratTx' = await executeTx(callAttempts, 'approveStratTx', Controller.approveStrategy, txRefs.'want', txRefs.'strategy'.address);
```

Comment:
Yeah I'm really starting to think 'want' and 'strategy' should just be local variables now. I can't figure out why you've grouped them in `txRefs` at all.